code/workspaces/web-app/src/components/stacks/StackCardActions/StackCardActions.js

```diff
@@ -115,15 +130,15 @@ export const PureStackCardActions = ({ stack, openStack, deleteStack, editStack,
           Share
         </StackMoreMenuItem>
         <StackMoreMenuItem
-          shouldRender={stackTypes.stackInCategory(stack.type, ANALYSIS, PUBLISH) && ownsStack}
+          shouldRender={userActions.restart && restartStack && ownsStack}
           onClick={() => restartStack(stack)}
           userPermissions={userPermissions}
           requiredPermission={editPermission}
         >
           Restart
         </StackMoreMenuItem>
         <StackMoreMenuItem
-          shouldRender={deleteStack && ownsStack}
+          shouldRender={userActions.delete && deleteStack && ownsStack}
```

Comment:
Similar to storage I think we rely on this component for projects as well, does this still get resolved correctly if we're relying on allowable actions being specified by the configmap?

Reply:
Good question: we do and it does. By default, if the specified type isn't found in the config map all the actions default to `true`. This then means the `shouldRender` falls back to whether the relevant functions are provided to the stack cards or not which is why the projects and storage currently works.

When we implement the new card design I think we should be aiming to create some more components so there is one for project, another for notebooks etc. which all reuse the same building blocks such that it makes changes like this much more clear and each component can be tailored to how the item they are representing works.